vignettes/irecist.Rmd

```diff
@@ -0,0 +1,914 @@
+patient has multiple observations per visit (e.g. by selecting the worst value).
+Another consideration could be extra potential protocol-specific sources of
+Progressive Disease such as radiological assessments, which could be
+pre-processed here to create a PD record to feed downstream derivations.
+
+For the derivation of the parameters it is expected that the subject
+identifier variables (usually `STUDYID` and `USUBJID`) and `ADT` are a
+unique key. This can be achieved by deriving an analysis flag
+(`ANLzzFL`). See [Derive `ANL01FL`](#anl01fl) for an example.
+
+The below shows an example of a possible company-specific implementation
+of this step.
+
+### Select Overall Response Records and Set Parameter Details
+
+In this case we use the overall response records from `RS` from the
+investigator as our starting point. The parameter details such as
+`PARAMCD`, `PARAM` etc will always be company-specific, but an example
+is shown below so that you can trace through how these records feed into
+the other parameter derivations.
+
+```{r}
+adrs <- adrs %>%
+  filter(RSEVAL == "INVESTIGATOR" & RSTESTCD == "OVRLRESP") %>%
+  mutate(
+    PARAMCD = "OVR",
+    PARAM = "Overall Response by Investigator",
+    PARCAT1 = "Tumor Response",
+    PARCAT2 = "Investigator",
+    PARCAT3 = "iRECIST"
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, VISIT, RSTESTCD, RSEVAL, PARAMCD, PARAM, PARCAT1, PARCAT2, PARCAT3)
+)
+```
+
+### Partial Date Imputation and Deriving `ADT`, `ADTF`, `AVISIT` etc
+
+If your data collection allows for partial dates, you could apply a
+company-specific imputation rule at this stage when deriving `ADT`. For
+this example, here we impute missing day to last possible date.
+
+```{r}
+adrs <- adrs %>%
+  derive_vars_dt(
+    dtc = RSDTC,
+    new_vars_prefix = "A",
+    highest_imputation = "D",
+    date_imputation = "last"
+  ) %>%
+  mutate(AVISIT = VISIT)
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, AVISIT, PARAMCD, PARAM, RSSTRESC, RSDTC, ADT, ADTF)
+)
+```
+
+### Derive `AVALC` and `AVAL`
+
+Here we populate `AVALC` and create the numeric version as `AVAL`
+(ordered from worst to best response). The `AVAL` values are not considered in
+the parameter derivations below, and so changing `AVAL` here would not change
+the result of those derivations. However, please note that the ordering of `AVAL` 
+will be used to determine `ANL01FL` in the subsequent step, ensure that the appropriate 
+`mode` is being set in the `admiral::derive_var_extreme_flag()`. 
+
+iRECIST ordering will be used or if you'd like to provide your own company-specific ordering here you
+could do this as follows:
+
+```{r}
+aval_resp_new <- function(arg) {
+  case_when(
+    arg == "iCR" ~ 8,
+    arg == "iPR" ~ 7,
+    arg == "iSD" ~ 6,
+    arg == "NON-iCR/NON-iUPD" ~ 5,
+    arg == "iUPD" ~ 4,
+    arg == "iCPD" ~ 3,
+    arg == "NE" ~ 2,
+    arg == "MISSING" ~ 1,
```

Comment:
Feedback from Uwe:
> Question on Derive AVAL and AVALC: is this ordering worst to best? - for me iCPD is worst

Reply:
I agree, what if we have NE and iCPD at the same day? I see similar approach with Recist 1.1 vignette. 
Maybe "ordered from worst to best response, followed by NE and MISSING"?

Reply:
I have not found an answer. I would treat NE as a missing value and would place it after iCR:
"MISSING" - 8,
"NE" - 7,
"iCR" - 6,
"iPR" - 5,
"iSD" - 4,
 "NON-iCR/NON-iUPD" - 3,
 "iUPD" - 2,
 "iCPD" - 1

Reply:
Thank you for following up. Will update as suggested.